packages/material-ui/src/MenuItem/MenuItem.js

```diff
@@ -110,10 +211,30 @@ MenuItem.propTypes = {
    * If `true`, the left and right padding is removed.
    */
   disableGutters: PropTypes.bool,
+  /**
+   * @ignore
+   */
+  handleArrowRightKeydown: PropTypes.func,
```

Comment:
These should follow the `on*` convention. Though I'm not sure why we need a separate handler for a specific key.

Reply:
Updated to the `on*` convention locally.

The main reason this is a super-specific handler is that it's provided to the `MenuItem` by the parent 'Menu' to allow a child `MenuItem` to affect the state of its parent `Menu` so the parent `Menu` can correctly orchestrate which submenu is open. ArrowRight (in RTL) is a fairly special action that only does something on `MenuItems` which have a `subMenu`.

Maybe there's an argument to make it a more generic `onKeydown` and just have it only know how to handle `ArrowRight` for now. But that's all this handler, passed from the parent, does at the moment and it's relatively special nature had me naming it special as well, when I wrote it initially.

Reply:
Could whatever passes the `handleArrowRightKeydown` pass `onKeyDown` instead and then handle `event.key === ArrowLeft`?

Reply:
Probably. I'll revisit this after I get a green CI.

Reply:
So, I can't pass `onKeydown` instead of `onArrowRightKeydown` unless we don't want users to be able to pass their own `onKeydown` handler and have that work as well. I forgot that was another reason for the weird naming: to avoid a collision with user-provided functionality.

I think we want to allow that pass through, especially since this is a MenuItem and cascading menus is not the only use case for this component. So, do you have any other preferences or conventions you'd like to follow on naming this prop?

Instead of naming it based on what it handles, we could name it based on where it's coming from, if that sounds better. I have no strong preference or sense of convention around this.

Reply:
> I forgot that was another reason for the weird naming: to avoid a collision with user-provided functionality.

We do this a lot in our components by composing user functions. The codebase should have plenty of examples for that.

Reply:
Yeah, I'm familiar with the usual pattern/convention but, in this case, the `onArrowRightKeydown` and the user-provided `onKeydown` are coming in on the same interface because of the cloning going on, so it doesn't seem like there's an opportunity to alias mine to something like `onKeydownProp` and then compose, like usual, unless I'm missing something.

Maybe I could pass all of the additional props that are being cloned onto each item as an object to a single prop and destructure it on the other side for consumption, like what's done for `TransitionProps` on `Menu` to get stuff into the TransitionProps of the `Popover`.